File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.ts
function topKFrequent(nums: number[], k: number): number[] {
    let occurences = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const current = nums[i];
        const occurence = occurences.get(current) || 0;

        occurences.set(current, occurence + 1); 
    }

    const result = Array.from(occurences.entries()).sort((a, b) => {
        return b[1] - a[1];
    });

    return result.slice(0, k).map(pair => pair[0]);
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.swift
class Solution {
    func topKFrequent(_ nums: [Int], _ k: Int) -> [Int] {
        var occurences: [Int: Int] = [:]

        for num in nums {
            if let occurence = occurences[num] {
                occurences[num] = occurence + 1
            } else {
                occurences[num] = 1
            }
        }

        var result = occurences.keys.sorted { occurences[$0]! < occurences[$1]! }

        var answer: [Int] = []
        var k = k

        while k > 0 {
            answer.append(result.removeLast())
            k -= 1
        }

        return answer
    }
}